apis/management.cattle.io/v3/authn_types.go

```diff
@@ -162,6 +162,27 @@ type GithubConfigApplyInput struct {
 	Enabled      bool         `json:"enabled,omitempty"`
 }
 
+type GoogleOAuthConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	AuthConfig        `json:",inline" mapstructure:",squash"`
+
+	ClientCredentialsFile string `json:"clientCredentialsFile,omitempty" norman:"required"`
+	ServiceAccountFile    string `json:"serviceAccountFile,omitempty"`
+	AdminEmail            string `json:"adminEmail,omitempty"`
```

Comment:
What admin?  Is this not already a part of one of the 2 credential JSON blobs above?

Reply:
This is the admin of the hosted google domain. I've seen that it's required in making api calls when using service account (which is also created by this admin). And it's not a part of the above 2 json files

Reply:
This is the admin of the hosted google domain. I’ve seen that it’s required in making api calls when using service account (which is also created by this admin). And it’s not a part of the above 2 json files
Ref: https://developers.google.com/admin-sdk/directory/v1/guides/delegation

Reply:
Does the `client_email` in the credential work?  It doesn't really make sense to have to specify who the key is for again...

Reply:
@vincent99 Even I thought it'd work but it didn't, I'll try another time to be sure, but even in the Go example they have given in this doc they use some other user email (admin of the domain) and not the client email.

Reply:
I still don't know what this means... Presumably there can be more than one admin of the domain.  Does it have to be a particular one?  Any one?  The same as the one that's making the service account? 

What if they are no longer an admin or part of the domain tomorrow?

Reply:
The service account needs to impersonate a user that has access to the admin apis, hence the admin of the domain. So this is the admin user who is creating the service account.
```
Note: Only users with access to the Admin APIs can access the Admin SDK Directory API, therefore your service account needs to impersonate one of those users to access the Admin SDK Directory API. Additionally, the user must have logged in at least once and accepted the G Suite Terms of Service.
```